### PR TITLE
Add pause button

### LINKED ISSUE #137: Add button to stop/restart watch
The lib ~~node-tail~~ tail enables to start/stop watch:

```
tail.watch();

tail.unwatch();
```

It's very good for many logs in little time when necessary analyse the log, so watch stop and after, start watch again.

It would be interesting to add this feature in this project as well.

I have an application that generates many logs.

### PR DESCRIPTION
**Pull request by levonet**
resolve #137
Знімок екрана 2019-04-11 о 15 23 42Знімок екрана 2019-04-11 о 15 23 49
Знімок екрана 2019-04-11 о 15 22 57Знімок екрана 2019-04-11 о 15 23 20
photo_2019-04-11 15 18 10

### CODE REVIEW

**Comment by levonet** [comment]
@mthenw Please see this PR

**Comment by levonet** [comment]
@mthenw ping

**Comment by mthenw** [comment]
Hey, I'm sorry about super late answer. What I actually see in this PR is a "ignore new lines" functionality. Could you elaborate more about what's the use case and are there any reported issues that this PR solves?

**Comment by levonet** [comment]
We have a large stream of logs and use filters.
First, it is difficult to keep a part of logs in view when new entries are added.
Second, the new data quickly displaces the necessary data from the buffer. We can't endlessly increase the number of rows that are stored in the browser.
Third, there is often a situation when you need to quickly stop the log in order not to search for the necessary information for a long time.

**Comment by RickeyWard** [comment]
As a frequent frontail user I'm in support of this functionality, but I'd like to see it tell you how many new log lines you've missed since it's been paused. and leave that in the list when it's unpaused so it's clear that there is a gap in the data you're seeing.

**Comment by levonet** [comment]
@RickeyWard maybe need to add a line with counter after a pause if it's possible.
Line like `==> SKIPED: 999 <==`.

**Comment by levonet** [comment]
@mthenw What prevents you from accepting my contribution?

**Comment by mthenw** [comment]
@levonet sorry for being so terrible at answering. I'm basically not sure if this is the type of functionality that I want to include in frontail. I don't have any strong arguments, tho :(

Let's do this. I'm testing this functionality right now and will give you some feedback shortly.

File: web/assets/app.js [hunk 1]
```diff
@@ -223,6 +242,17 @@ window.App = (function app(window, document) {
...
       });
 
+      // Paused button bind
+      _pausedBtn.addEventListener('mouseup', function() {
+        _isPaused = !_isPaused;
+        if (_isPaused) {
+          this.className += ' active';
+        } else {
+          _skipCounter = 0;
+          this.classList.remove("active");
```
Comment (mthenw) [inline web/assets/app.js]:
```suggestion
          this.classList.remove('active');
```

File: test/app.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ describe('browser application', () => {
       socket: io,
       container: window.document.querySelector('.log'),
       filterInput: window.document.querySelector('#filter'),
+      pausedBtn: window.document.querySelector('#pausedBtn'),
```
Comment (mthenw) [inline test/app.js]: Can you rename it to `pauseBtn`? Both variables and id.

File: web/assets/styles/dark.css [hunk 3]
```diff
@@ -19,6 +19,17 @@ body {
   color: #999;
 }
 
+.btn-paused {
+    background: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' fill='%237f8289' viewBox='0 0 8 8'><path d='M1 1v6h2v-6h-2zm4 0v6h2v-6h-2z'></path></svg>") no-repeat center center;
+    background-color: #2f3238;
+    border: 1px solid transparent;
+}
+
+.btn-paused.active {
```
Comment (mthenw) [inline web/assets/styles/dark.css]:
```suggestion
.btn-pause.play {
```

**Comment by mthenw** [comment]
@levonet let me know if you are still willing to contribute (you already did a lot and I totally get it if you're no longer interested).

If not I will fix those few issues by myself?

**Comment by levonet** [comment]
> In the current implementation favicon badge updates as if there're new lines. But when the streaming is paused we shouldn't probably update it. Can you fix that?

This problem occurs when the filter input loses focus.
Apparently, this problem was before my changes. I'll try to fix it.

**Comment by levonet** [comment]
@mthenw done

**Comment by mthenw** [comment]
@levonet thank you for your contribution and sorry that it took so long.

**Pull request merged**